[FriendlyUI only] Add notification for when posts are added to a sequence

s-cheng opened the pull request:
This PR adds the ability for users to get notified when posts are added to a sequence. I've only enabled it for friendly sites because there are some UI additions, including a new email.

New button on sequence page:
<img width="746" alt="Screen Shot 2024-03-30 at 9 05 51 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/ffea5590-c0d2-430c-bbc3-f69c1bde592e">

New on-site notification:
<img width="715" alt="Screen Shot 2024-03-30 at 9 08 21 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/0948524f-2f2e-4934-b5a5-ba5426d567fb">

New email notification (with an image from the sequence so it's less boring - yes this is a poster for the off-brand spice girls):
<img width="535" alt="Screen Shot 2024-03-30 at 10 05 31 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/d8bab54b-2152-427d-b6ed-d06d08c1f3a7">

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206971521676313) by [Unito](https://www.unito.io)

s-cheng commented on packages/lesswrong/components/form-components/ImageUpload.tsx:
```diff
@@ -67,15 +68,16 @@ const styles = (theme: ThemeType): JssStyles => ({
-    width: 203,
-    height: 80
+    width: 250,
+    height: 100
```
This just makes the card image preview the same size as the sequence card image

s-cheng commented on packages/lesswrong/components/form-components/ImageUpload.tsx:
```diff
@@ -67,15 +68,16 @@ const styles = (theme: ThemeType): JssStyles => ({
-    height: 280
+    width: 1600,
+    height: 380,
+    imgProps: {g: 'custom', dpr: '2.0'}
```
This makes the banner image preview actually look the same as the sequence banner image on the page (right now it's positioned differently and stretched out)

s-cheng commented on packages/lesswrong/components/notifications/NotifyMeButton.tsx:
```diff
@@ -97,23 +108,28 @@ const NotifyMeButton = ({
+    ) : (
+      <button onClick={onSubscribe} className={classNames(className, classes.root)}>
+        {button}
+      </button>
```
I think the `asButton` version was only used in the post author card anyway, which is no longer on any site

s-cheng commented on packages/lesswrong/components/sequences/SequencesPage.tsx:
```diff
@@ -38,7 +57,7 @@ const styles = (theme: ThemeType): JssStyles => ({
     marginTop: theme.spacing.unit * 2,
     marginLeft: theme.spacing.unit/2,
-    marginBottom: theme.spacing.unit * 2,
+    marginBottom: isFriendlyUI ? 40 : theme.spacing.unit * 2,
```
Also added more space between the sequence description and the post list

s-cheng commented on packages/lesswrong/components/users/ViewSubscriptionsPage.tsx:
```diff
@@ -217,6 +218,14 @@ const ViewSubscriptionsPage = ({classes}: {
+      fragmentName="SequencesPageTitleFragment"
+      renderDocument={(sequence: SequencesPageTitleFragment) => <Link to={sequenceGetPageUrl(sequence)}>{sequence.title}</Link>}
+      noSubscriptionsMessage="You are not subscribed to any sequences."
+    />}
```
I might clean up this page in a separate PR, to move away from using "subscription" since that can be confusing

s-cheng commented on packages/lesswrong/server/repos/PostsRepo.ts:
```diff
@@ -97,7 +97,8 @@ class PostsRepo extends AbstractRepo<"Posts"> {
         p."authorIsUnreviewed" is not true AND
-        p."draft" is not true
+        p."draft" is not true AND
+        p."deletedDraft" is not true
```
Small fix for the digest planner page

jpaddison3 commented on packages/lesswrong/lib/collections/users/schema.ts:
```diff
@@ -1383,6 +1383,11 @@ const schema: SchemaType<"Users"> = {
     label: "Posts added to tags I'm subscribed to",
     ...notificationTypeSettingsField(),
   },
+  notificationSubscribedSequencePost: {
+    label: "Posts added to sequences I'm subscribed to",
+    ...notificationTypeSettingsField({ channel: "both" }),
+    hidden: !allowSubscribeToSequencePosts
+  },
```
I agree this is the idiomatic way to do this, but I do sure wish there was broader umbrella we could use.

jpaddison3 commented:
I'm getting cannot read _id of undefined on the best of page. EABestOfPage L 232.

s-cheng commented:
> I'm getting cannot read _id of undefined on the best of page. EABestOfPage L 232.

@jpaddison3 I believe that's because we updated the hard-coded list of posts recently, so some of the new ones don't exist in the dev db. We should prob make sure that kind of thing doesn't break that page, so I went ahead and fixed that in a new commit.

jpaddison3 commented:
Is this how the notification preview menu is supposed to look?
<img width="312" alt="image" src="https://github.com/ForumMagnum/ForumMagnum/assets/10352319/dc2792a0-80b9-4b29-abc7-aef6e8cdfa7f">

s-cheng commented:
@jpaddison3 I think you need to refresh the page for the hoverover to be up-to-date. Or is this after a refresh?

jpaddison3 commented on packages/lesswrong/components/notifications/NotifyMeButton.tsx:
```diff
@@ -97,23 +108,28 @@ const NotifyMeButton = ({
-  const {MenuItem} = Components;
-  const maybeMenuItemButton = asMenuItem ?
+  // Determine if this component should look like a link, menu item, or button
+  let maybeMenuItemButton = asMenuItem ?
```
[Nit]

I'd make this default null and then set it in an if elseif else block

jpaddison3 commented:
Ahhh, that works

jpaddison3 reviewed (approved):
LGTM, good stuff
